src/app/recents/recents.component.spec.ts

```diff
@@ -0,0 +1,61 @@
+import { fakeAsync, ComponentFixture, TestBed } from '@angular/core/testing';
+import { TabsModule } from 'ngx-bootstrap/tabs';
+import { of } from 'rxjs';
+import { NavigationEnd, Router, ActivatedRoute } from '@angular/router';
+import { HttpClientTestingModule } from '@angular/common/http/testing';
+import { NoopAnimationsModule } from '@angular/platform-browser/animations';
+import { NO_ERRORS_SCHEMA } from '@angular/core';
+
+import { RecentsComponent } from './recents.component';
+import { RecentsService } from '../services/recents.service';
+import { recents } from './test.data';
+import { NodeTypeDisplayPipe } from '../pipes/nodedisplay';
+
+
+const url = 'http://localhost:4200/recents';
+class MockRouter {
+  public events = of(new NavigationEnd(0, url, url));
+}
+
+
+describe('RecentsComponent', () => {
+  let fixture: ComponentFixture<RecentsComponent>;
+  let recentsService: RecentsService;
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      declarations: [ RecentsComponent, NodeTypeDisplayPipe ],
+      providers: [
+        RecentsService,
+        {provide: Router, useClass: MockRouter},
+        {
```

Comment:
This is not required since you don't need to simulate a router event. This can be removed.